.github/workflows/s3-workflow-rollout-to-scaleway.yml

```diff
@@ -0,0 +1,69 @@
+---
+name: Rollout S3 buckets (Scaleway)
+on:  # yamllint disable-line rule:truthy
+  workflow_call:
+    inputs:
+      # Scaleway information
+      scaleway-region:
+        description: Scaleway region to deploy to
+        required: true
+        type: string
+    secrets:
+      SCALEWAY_ACCESS_KEY:
+        description: Scaleway API key ID
+        required: true
+      SCALEWAY_SECRET_KEY:
+        description: Scaleway API key secret
+        required: true
+
+jobs:
+  generate-matrix:
+    name: Generate job matrices
+    runs-on: ubuntu-latest
+    outputs:
+      buckets: ${{ steps.filtered.outputs.changes }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          sparse-checkout: |
+            s3
+            .github
+      - name: Generate Matrix
+        id: filtered
+        uses: dorny/paths-filter@v3
+        with:
+          filters: .github/s3-rollout-filters.yaml
+
+  rollout-buckets:
+    runs-on: ubuntu-latest
+    needs: [generate-matrix]
+    name: S3CMD
+    strategy:
+      matrix:
+        deployment: ${{ fromJson(needs.generate-matrix.outputs.buckets) }}
+    if: ${{ needs.generate-matrix.outputs.buckets != '[]' }}]
+    environment: ${{ matrix.deployment }}
+    steps:
+      - name: Checkout s3 folder
+        uses: actions/checkout@v4
+        with:
+          sparse-checkout: |
+            s3
+
+      - name: Set up S3cmd cli tool for environment
+        uses: s3-actions/s3cmd@main
+        with:
+          provider: scaleway
+          region: ${{ inputs.scaleway-region }}
+          access_key: ${{ secrets.SCW_ACCESS_KEY_ID }}
+          secret_key: ${{ secrets.SCW_SECRET_ACCESS_KEY }}
+
+      - name: Check existing/new buckets and apply lifecycle and cors
```

Comment:
Move this code into a separate rollout s3 job, which could be reused for other providers. Same as we did with docker, k8s, etc…